src/ragger/backend/speculos.py

```diff
@@ -152,3 +154,23 @@ def compare_screen_with_snapshot(self,
                                     lower=crop.lower)
         else:
             return screenshot_equal(f"{golden_snap_path}", snap)
+
+    def wait_for_screen_change(self, timeout: float = 10.0, context: list = []):
+        start = time()
+        if not context:
+            return self._client.get_screen_content()
+        else:
+            content = context
+            while content == context:
+                content = self._client.get_screen_content()
+                now = time()
+                if (now - start > timeout):
+                    raise TimeoutError("Timeout waiting for screen change")
+                sleep(0.1)
+            return content
```

Comment:
```suggestion
            content = self._client.get_screen_content()
            while content == context:
                sleep(0.1)
                now = time()
                if (now - start > timeout):
                    raise TimeoutError("Timeout waiting for screen change")
                content = self._client.get_screen_content()
            return content
```
Pour éviter d'attendre si ça ne sert à rien ?

Reply:
Ok.